crates/oxc_linter/src/rules/eslint/func_names.rs

```diff
@@ -0,0 +1,507 @@
+        ), // { "ecmaVersion": 6 },
+        ("var foo = bar(function *() {});", Some(serde_json::json!(["never"]))), // { "ecmaVersion": 6 },
+        ("var foo = function*() {};", Some(serde_json::json!(["never"]))), // { "ecmaVersion": 6 },
+        ("(function*() {}())", Some(serde_json::json!(["never"]))),        // { "ecmaVersion": 6 },
+        (
+            "var foo = bar(function *() {});",
+            Some(serde_json::json!(["never", { "generators": "never" }])),
+        ), // { "ecmaVersion": 6 },
+        (
+            "var foo = function*() {};",
+            Some(serde_json::json!(["never", { "generators": "never" }])),
+        ), // { "ecmaVersion": 6 },
+        ("(function*() {}())", Some(serde_json::json!(["never", { "generators": "never" }]))), // { "ecmaVersion": 6 },
+        (
+            "var foo = bar(function *() {});",
+            Some(serde_json::json!(["always", { "generators": "never" }])),
+        ), // { "ecmaVersion": 6 },
+        (
+            "var foo = function*() {};",
+            Some(serde_json::json!(["always", { "generators": "never" }])),
+        ), // { "ecmaVersion": 6 },
+        ("(function*() {}())", Some(serde_json::json!(["always", { "generators": "never" }]))), // { "ecmaVersion": 6 },
+        (
+            "var foo = bar(function *() {});",
+            Some(serde_json::json!(["as-needed", { "generators": "never" }])),
+        ), // { "ecmaVersion": 6 },
+        (
+            "var foo = function*() {};",
+            Some(serde_json::json!(["as-needed", { "generators": "never" }])),
+        ), // { "ecmaVersion": 6 },
+        ("(function*() {}())", Some(serde_json::json!(["as-needed", { "generators": "never" }]))), // { "ecmaVersion": 6 },
+        ("class C { foo = function() {}; }", Some(serde_json::json!(["as-needed"]))), // { "ecmaVersion": 2022 },
+        ("class C { [foo] = function() {}; }", Some(serde_json::json!(["as-needed"]))), // { "ecmaVersion": 2022 },
+        ("class C { #foo = function() {}; }", Some(serde_json::json!(["as-needed"]))), // { "ecmaVersion": 2022 }
+    ];
+
+    let fail = vec![
+        ("Foo.prototype.bar = function() {};", None),
+        ("(function(){}())", None),
+        ("f(function(){})", None),
+        ("var a = new Date(function() {});", None),
+        ("var test = function(d, e, f) {};", None),
+        ("new function() {}", None),
+        ("Foo.prototype.bar = function() {};", Some(serde_json::json!(["as-needed"]))),
+        ("(function(){}())", Some(serde_json::json!(["as-needed"]))),
+        ("f(function(){})", Some(serde_json::json!(["as-needed"]))),
+        ("var a = new Date(function() {});", Some(serde_json::json!(["as-needed"]))),
+        ("new function() {}", Some(serde_json::json!(["as-needed"]))),
+        ("var {foo} = function(){};", Some(serde_json::json!(["as-needed"]))), // { "ecmaVersion": 6 },
+        // ("({key: foo = function(){}} = {});", Some(serde_json::json!(["as-needed"]))), // { "ecmaVersion": 6 },
```

Comment:
Does this need fixing?

Reply:
nice catch, this is a passing test, I just copied rfom above to test something, will remove it :)